dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AudioToText;
+using Microsoft.SemanticKernel.Contents;
+
+namespace Microsoft.SemanticKernel.Connectors.AssemblyAI;
+
+/// <summary>
+/// AssemblyAI speech-to-text service.
+/// </summary>
+[Experimental("SKEXP0005")]
+public sealed class AssemblyAIAudioToTextService : IAudioToTextService
+{
+    private readonly string _apiKey;
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Attributes is not used by AssemblyAIAudioToTextService.
+    /// </summary>
+    public IReadOnlyDictionary<string, object?> Attributes => new Dictionary<string, object?>();
+
+    /// <summary>
+    /// Creates an instance of the <see cref="AssemblyAIAudioToTextService"/> with an AssemblyAI API key.
+    /// </summary>
+    /// <param name="apiKey">OpenAI API Key</param>
+    /// <param name="httpClient"></param>
+    public AssemblyAIAudioToTextService(
+        string apiKey,
+        HttpClient httpClient
+    )
+    {
+        this._apiKey = apiKey;
+        this._httpClient = httpClient;
+    }
+
+    /// <summary>
+    /// Transcribe audio file.
+    /// </summary>
+    /// <param name="content">Audio content.</param>
+    /// <param name="executionSettings">The AI execution settings (optional).</param>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Text content from audio content.</returns>
+    public async Task<TextContent> GetTextContentAsync(
+        AudioContent content,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        string uploadUrl;
+        using var stream = content.Data!.ToStream();
+        {
+            uploadUrl = await this.UploadFileAsync(stream, cancellationToken).ConfigureAwait(false);
+        }
+
+        var transcriptId = await this.CreateTranscriptAsync(uploadUrl, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+        var transcript = await this.WaitForTranscriptToProcessAsync(transcriptId, cancellationToken)
+            .ConfigureAwait(false);
+
+        return new TextContent(
+            text: transcript.RootElement.GetProperty("text").GetString(),
+            modelId: null,
```

Comment:
modelId is always null?

Reply:
Our customers are automatically on our latest and best model.
We do have a lightweight model coming in which case the `modelId` parameter may be useful for us, because the customer will be able to opt for this new model by its name.